TumbleBitSetup/RsaPubKey.cs

```diff
@@ -31,7 +31,17 @@ public RsaPubKey(BigInteger N, BigInteger e)
         {
             _pubKey = new RsaKeyParameters(false, N, e);
         }
-        
+
+        public bool VerifyPoupardSternProof(PoupardSternProof proof, PoupardSternSetup setup)
+        {
+            return PoupardStern.Verifying(this, proof.XValues, proof.YValue, setup.KeySize, setup.PublicString, setup.K);
```

Comment:
Yes, we are passing keySize from the setup, not deriving it from the pubkey. Good.